File: java/UVA10170.java
//https://zerojudge.tw/ShowProblem?problemid=e555
import java.util.Scanner;

class Main{
	public static void main(String args[]){
		Scanner sc=new Scanner(System.in);
		
		while(sc.hasNext()){
			long S=sc.nextLong();
			long D=sc.nextLong();
			
			long Stemp=S;
			long count=S;
			while(Stemp<D){
				count++;
				Stemp=Stemp+count;
			}
			
			System.out.println(count);
		}
	}
}

File: cpp/UVA10170.cpp
//https://zerojudge.tw/ShowProblem?problemid=e555
#include <bits/stdc++.h>
using namespace std;

int main() {
    long long S, D;

    while (cin >> S >> D) {
        long long sum = 0;

        while (sum < D) {
            sum += S;
            S++;
        }

        cout << S - 1 << '\n';
    }

    return 0;
}